testrunner/src/mill/testrunner/TestRunnerUtils.scala

```diff
@@ -213,6 +234,108 @@ import scala.jdk.CollectionConverters.IteratorHasAsScala
     (doneMessage, results.toSeq)
   }
 
+  private def stealTaskFromSelectorFolder(
+      globSelectorCache: Map[String, ClassWithFingerprint],
+      runner: Runner,
+      stealFolder: os.Path,
+      selectorFolder: os.Path
+  ): Array[Task] = {
+    val stealLog = stealFolder / os.up / "steal.log"
+    while (true) {
+      val files = os.list(selectorFolder)
```

Comment:
I think we can simplify this further given the current design: we just need to run `os.list` once and then loop over all the tests to try and steal each one in order. Since the files in `selectorFolder` never grows, running `os.list` a second time will never pick up anything the first list didn't.

Let's also get rid of the `Random.nextInt` selection and just pick up the tests in order from top to bottom. That will give a bit more determinism to how things run, and although it still nondeterministic due to parallelism that would help people follow the order that tests are being run in

Reply:
The `os.move` already effectively checks whether the file is still left before taking it, so we don't need to run an entire `os.list` each time we steal a single file. We could add a `os.exists` check before `os.move` if we want to avoid throwing and catching an error in the common case